x-pack/plugins/rule_registry/server/utils/get_common_alert_fields.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Values } from '@kbn/utility-types';
+import { AlertExecutorOptions } from '../../../alerting/server';
+import { ParsedTechnicalFields } from '../../common/parse_technical_fields';
+import {
+  ALERT_ID,
+  ALERT_UUID,
+  ALERT_RULE_CATEGORY,
+  ALERT_RULE_CONSUMER,
+  ALERT_RULE_NAME,
+  ALERT_RULE_PRODUCER,
+  ALERT_RULE_TYPE_ID,
+  ALERT_RULE_UUID,
+  SPACE_IDS,
+  TAGS,
+  TIMESTAMP,
+} from '../../common/technical_rule_data_field_names';
+
+const commonAlertFieldNames = [
+  ALERT_RULE_CATEGORY,
+  ALERT_RULE_CONSUMER,
+  ALERT_RULE_NAME,
+  ALERT_RULE_PRODUCER,
+  ALERT_RULE_TYPE_ID,
+  ALERT_RULE_UUID,
+  SPACE_IDS,
+  TAGS,
+  TIMESTAMP,
+];
+export type CommonAlertFieldName = Values<typeof commonAlertFieldNames>;
+
+const commonAlertIdFieldNames = [ALERT_ID, ALERT_UUID];
+export type CommonAlertIdFieldName = Values<typeof commonAlertIdFieldNames>;
+
+export type CommonAlertFields = Pick<ParsedTechnicalFields, CommonAlertFieldName>;
+
+export const getCommonAlertFields = (
+  options: AlertExecutorOptions<any, any, any, any, any>
+): CommonAlertFields => {
+  return {
+    [ALERT_RULE_CATEGORY]: options.rule.ruleTypeName,
```

Comment:
`category` would imply that multiple rule types could be grouped by having the same value in this field, but `ruleTypeName` is specific to the rule type. Should we make a new field `kibana.alert.rule.rule_type_name` to store this value?

Reply:
Thanks for weighing in. On the one hand that sounds reasonable. OTOH I'm not convinced this piecemeal approach of adjusting individual field names as we find them is efficient enough given the time constraints. Would you be ok if we deferred this to a follow-up (maybe #109033) so we can get this merged as soon as possible?